ocaml/typing/typecore.ml

```diff
@@ -7780,6 +7790,9 @@ let report_error ~loc env = function
        "@[The constructor %a@ expects %i argument(s),@ \
         but is applied here to %i argument(s)@]"
        longident lid expected provided
+  | Constructor_labeled_arg ->
+      Location.errorf ~loc
+       "Constructors cannot receive labeled arguments"
```

Comment:
I agree we don't want to support this (at least, for now) and that rejecting it here is easier than rejecting it in the parser.

Maybe the error message could give a little hint, though:   "Constructors cannot receive labeled arguments.  Use an inline record instead."